packages/NodeTypeResolver/Reflection/BetterReflection/SourceLocatorProvider/DynamicSourceLocatorProvider.php

```diff
@@ -64,28 +70,18 @@ public function provide(): SourceLocator
             $sourceLocators[] = new OptimizedSingleFileSourceLocator($this->fileNodesFetcher, $file);
         }
 
-        foreach ($this->filesByDirectory as $files) {
-            $sourceLocators[] = new OptimizedDirectorySourceLocator($this->fileNodesFetcher, $this->phpVersion, $files);
+        foreach ($this->directories as $directory) {
```

Comment:
@TomasVotruba I tested this in CodeIgniter4 project, it make consistency 1 second faster.

Also make correct apply `::class` tested at:

- https://github.com/codeigniter4/CodeIgniter4/pull/7847

I will check more if this while improve performance, on complex files in target dir, eg contains `.xml` file, it read unnecessary files, since it fetch files under directory instead of read only listed files in the directory.

Reply:
@TomasVotruba I debugged, the `OptimizedDirectorySourceLocatorFactory` utilize PHPStan `FileFinder` that already only cover `php` extensions, so I guess it already cover only scan php files.

```php
PHPStan\File\FileFinder #3115
   fileExcluder: PHPStan\File\FileExcluder #3113
   fileExtensions: array (1)
   |  0 => 'php'
```

so I think it already ok 👍